runtimebp/cpu.go

```diff
@@ -127,11 +134,108 @@ func defaultMaxProcsFormula(n float64) int {
 	return int(math.Ceil(n))
 }
 
-// GOMAXPROCS sets runtime.GOMAXPROCS with the default formula,
+func scaledMaxProcsFormula(n float64) int {
+	// If we have a request below 1, round up so we don't end up single-threaded
+	if n < 1 {
+		n = 1
+	}
+	i := fetchMaxProcsMult()
+	return int(math.Ceil(n * float64(i)))
+}
+
+func fetchMaxProcsMult() int {
```

Comment:
```suggestion
func fetchMaxProcsMultiplier() int {
```
:no_bell: shorter environment makes sense but maybe to make it easier to understand that it is a factor.